SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1950,14 +1950,29 @@ ERPCResult USpatialReceiver::ApplyRPCInternal(UObject* TargetObject, UFunction*
 	TSharedPtr<FRepLayout> RepLayout = NetDriver->GetFunctionRepLayout(Function);
 	RepLayout_ReceivePropertiesForRPC(*RepLayout, PayloadReader, Parms);
 
-	if ((UnresolvedRefs.Num() == 0) || bApplyWithUnresolvedRefs)
+	if (UnresolvedRefs.Num() != 0 && !bApplyWithUnresolvedRefs)
 	{
-		TargetObject->ProcessEvent(Function, Parms);
-		Result = ERPCResult::Success;
+		ErrorInfo.ErrorCode = ERPCResult::UnresolvedParameters;
 	}
 	else
 	{
-		Result = ERPCResult::UnresolvedParameters;
+		// Get the RPC target Actor.
+		AActor* Actor = Cast<AActor>(TargetObject);
+		if (Actor == nullptr)
+		{
+			Actor = Cast<UActorComponent>(TargetObject)->GetOwner();
+		}
```

Comment:
This will not work for subobjects that aren't `UActorComponent`s. I believe this is what we use in other places:
```suggestion
		AActor* Actor = TargetObject->GetTypedOuter<AActor>();
```